.github/workflows/workflow_call.yml

```
@@ -0,0 +1,61 @@
+name: Integration Tests (Workflow Call)
+on:
+  workflow_dispatch:
+    inputs:
+      dd-trace-go-ref:
+        description: 'The ref to checkout dd-trace-go at'
+        required: false
+        type: string
+        default: main
+  workflow_call:
+    inputs:
+      dd-trace-go-ref:
+        type: string
+        required: true
+        description: 'The ref to checkout dd-trace-go at'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

`github.event.pull_request.number` is never going to be set; because this does not trigger from `pull_request`. And `github.ref` is (I think) going to be the current tip of `main` on this repository... This means multiple PRs from a remote repo would cancel each other.

Probably this would be better:
```suggestion
  group: ${{ github.workflow }}-${{ inputs.dd-trace-go-ref }}
```